### PR TITLE
mozwire: init at 0.4.1

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Add mozwire.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
@GrahamcOfBorg build mozwire

**Comment by siraben** [comment]
@Infinisil looks good to merge?

**Comment by siraben** [comment]
@GrahamcOfBorg build mozwire

File: pkgs/tools/networking/mozwire/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ pkgs, stdenv, fetchFromGitHub }:
+pkgs.rustPlatform.buildRustPackage rec {
```
Comment (zowoq) [inline pkgs/tools/networking/mozwire/default.nix:2]:
```suggestion
{ rustPlatform, stdenv, fetchFromGitHub }:
rustPlatform.buildRustPackage rec {
```

File: pkgs/tools/networking/mozwire/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "NilsIrl";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1slfb6m22vzglnrxahlhdcwzwpf3b817mskdx628s92mjzngzyih";
+  };
+
+  cargoSha256 = "0b00j8vn1vvvphcyv8li7i73pq66sq6dr4wc1w4s3pppa151xr55";
+  verifyCargoDeps = true;
```
Comment (zowoq) [inline pkgs/tools/networking/mozwire/default.nix:14]: This is option was removed.

```suggestion
```

File: pkgs/tools/networking/mozwire/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with stdenv.lib; {
+    description = "MozillaVPN configuration manager giving Linux, macOS users\
+(among others), access to MozillaVPN.";
```
Comment (zowoq) [inline pkgs/tools/networking/mozwire/default.nix:18]: No need to split this over multiple lines.

```suggestion
    description = "MozillaVPN configuration manager giving Linux, macOS users (among others), access to MozillaVPN";
```

File: pkgs/tools/networking/mozwire/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    homepage = https://github.com/NilsIrl/MozWire;
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ siraben nilsirl ];
+    platforms = platforms.all;
```
Comment (zowoq) [inline pkgs/tools/networking/mozwire/default.nix:22]: `buildRustPackage` sets `meta.platforms`.

```suggestion
```

File: pkgs/tools/networking/mozwire/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "1slfb6m22vzglnrxahlhdcwzwpf3b817mskdx628s92mjzngzyih";
+  };
+
+  cargoSha256 = "0b00j8vn1vvvphcyv8li7i73pq66sq6dr4wc1w4s3pppa151xr55";
+  verifyCargoDeps = true;
+
+  meta = with stdenv.lib; {
+    description = "MozillaVPN configuration manager giving Linux, macOS users\
+(among others), access to MozillaVPN.";
+    homepage = https://github.com/NilsIrl/MozWire;
```
Comment (zowoq) [inline pkgs/tools/networking/mozwire/default.nix:19]: `homepage` should be in quotes per https://github.com/NixOS/rfcs/pull/45.

```suggestion
    homepage = "https://github.com/NilsIrl/MozWire";
```

**Review by zowoq — COMMENTED** [review]
Please rename the commits:
```
mozwire: init at 0.4.1
maintainers: add nilsirl
```

**Comment by siraben** [comment]
@GrahamcOfBorg build mozwire

File: pkgs/tools/networking/mozwire/default.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{ rustPlatform, stdenv, fetchFromGitHub, darwin }:
```
Comment (zowoq) [inline pkgs/tools/networking/mozwire/default.nix:1]:
```suggestion
{ rustPlatform, stdenv, fetchFromGitHub, Security }:
```

File: pkgs/tools/networking/mozwire/default.nix [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  version = "0.4.1";
+
+  src = fetchFromGitHub {
+    owner = "NilsIrl";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1slfb6m22vzglnrxahlhdcwzwpf3b817mskdx628s92mjzngzyih";
+  };
+
+  buildInputs = stdenv.lib.optionals stdenv.isDarwin [ darwin.Security ];
```
Comment (zowoq) [inline pkgs/tools/networking/mozwire/default.nix:13]:
```suggestion
  buildInputs = stdenv.lib.optionals stdenv.isDarwin [ Security ];
```

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -3212,6 +3212,8 @@ in
 
   mcrcon = callPackage ../tools/networking/mcrcon {};
 
+  mozwire = callPackage ../tools/networking/mozwire {};
```
Comment (zowoq) [inline pkgs/top-level/all-packages.nix:3215]:
```suggestion
  mozwire = callPackage ../tools/networking/mozwire {
    inherit (darwin.apple_sdk.frameworks) Security;
  };
```

**Review by zowoq — COMMENTED** [review]
Please squash the darwin changes into the original commit.

**Review by zowoq — APPROVED** [review]
Builds/runs on darwin.

**Pull request merged**